Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+            end
+        end
+        function host= set_PB_host(obj,host,~)
+            try
+                obj.PulseBlaster = Drivers.PulseBlaster.instance(host);
+                obj.source_on = obj.PulseBlaster.lines(obj.PB_line).state;
+            catch
+                host = obj.no_server;
+            end
+        end
+        
+        function val = set_target_wavelength(obj, val, pref)   % Calls the tune() method.
+            if isnan(val); return; end % Short circuit on NaN
+            if obj.updatingVal; val = pref.value; warning('Laser is currently tuning.'); return; end
+            obj.tune(val);
+        end
+        
+        function val = set_etalon_percent(obj, val, ~)
+            if isnan(val); return; end % Short circuit on NaN
+            obj.getFrequency();
+            obj.do_etalon_lock = false;
+            obj.com('set_etalon_val', 'solstis', val);
+        end
+        function val = set_do_etalon_lock(obj, val, pref)
+            if val == pref.value; return; end
+            if obj.do_wavelength_lock && ~val   % Wavelength lock cannot be on without etalon lock.
+                obj.do_wavelength_lock = false;
+            end
+            obj.com('set_etalon_lock', 'solstis', obj.lockList{val+1});
+            obj.getFrequency();
+        end
+        
+        function val = set_resonator_percent(obj, val, ~)
+            if isnan(val); return; end % Short circuit on NaN
+            obj.getFrequency();
+            obj.do_wavelength_lock = false;
+            obj.com('set_resonator_val', 'solstis', val);
+        end
+        function val = set_do_wavelength_lock(obj, val, pref)
+            if val == pref.value; return; end
+            if ~obj.do_etalon_lock && val       % Wavelength lock cannot be on without etalon lock.
+                obj.do_etalon_lock = true;
+            end
+            obj.com('lock_wavelength', 'solstis', obj.lockList{val+1});
+            if val
+                try
+                    obj.tune(obj.setpoint_) % If this line is neglected, the laser will lock to whatever the current value is (which often is slightly off the setpoint).
+                catch
+                    obj.getFrequency();
+                end
+            else
+                obj.getFrequency();
+            end
+        end
+    end
+    
+    methods     % TunableLaser_invisible methods
+        function TuneCoarse(obj, target)
+            obj.TuneSetpoint(target);
+            obj.do_wavelength_lock = false;
+        end
+        function TunePercent(obj, target)
+            assert(target >= 0 && target <= 100, 'Target must be a percentage')
+            obj.resonator_percent = target;
+        end
+        function TuneSetpoint(obj, target)
+            obj.tune(obj.c/target);
+        end
+        function percent = GetPercent(obj)
+            obj.getFrequency();
+            obj.resonator_percent = obj.resonator_voltage/2;
```

Comment:
Should the 2 here be a constant property of the laser?

Reply:
I don't think it matters too much, but maybe.